src/backend/executor/execMain.c

```diff
@@ -1214,12 +1233,21 @@ standard_ExecutorEnd(QueryDesc *queryDesc)
         cdbexplain_sendExecStats(queryDesc);
 
 	/*
-	 * if needed, collect mpp dispatch results and tear down
-	 * all mpp specific resources (e.g. interconnect).
+	 * if needed, tear down all mpp specific resources (e.g. interconnect).
 	 */
 	PG_TRY();
 	{
-		mppExecutorFinishup(queryDesc);
+		/*
+		 * Check and free the results of all gangs. If any QE had an
+		 * error, report it and exit to our error handler via PG_THROW.
+		 * NB: This call doesn't wait, because we already waited previously.
+		 */
```

Comment:
Agree, that this comment is some kind of rudiment. [Here](https://github.com/greenplum-db/gpdb/commit/957629d145b42793b6706a72e0d076b51de3255a#diff-b749055c948314574d0e9d9a67a9f13041d7a050064ffae75be875e5a3cbfd5fL1174-L1179), in the original commit that removed `cdbdisp_finishCommand`, the situation is quite the same.
Thus, I guess, PG_TRY is also redundant